libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -189,73 +265,168 @@ export class BotConfiguration extends BotConfigurationBase {
         }
     }
 
-    // save the config file to specificed botpath
-    public async saveAs(botpath: string, secret?: string): Promise<void> {
-        if (!botpath) {
-            throw new Error(`missing path`);
-        }
-
-        this._savePrep(secret);
-
-        const hasSecret: boolean = !!this.secretKey;
-
-        if (hasSecret) {
-            this.encrypt(secret);
-        }
-        await fsx.writeJson(botpath, this.toJSON(), { spaces: 4 });
-
-        if (hasSecret) {
-            this.decrypt(secret);
-        }
-    }
-
-    // save the config file to specificed botpath
-    public saveAsSync(botpath: string, secret?: string): void {
-        this._savePrep(secret);
+    // export the services from the bot file as resource files and recipe file
+    public async export(folder: string, progress?: (service: IConnectedService, command: string, index: number, total: number) => void): Promise<BotRecipe> {
+        let recipe = new BotRecipe();
 
-        const hasSecret: boolean = !!this.secretKey;
+        await fsx.ensureDir(folder);
 
-        if (hasSecret) {
-            this.encrypt(secret);
-        }
+        let index = 0;
+        for (let service of this.services) {
+            index++;
 
-        fsx.writeJsonSync(botpath, this.toJSON(), { spaces: 4 });
-
-        if (hasSecret) {
-            this.decrypt(secret);
-        }
-    }
-
-    // save the config file back over original
-    public async save(secret?: string): Promise<void> {
-        return this.saveAs(this.internal.location, secret);
-    }
-
-    // save the config file back over original (blocking)
-    public saveSync(secret?: string): void {
-        return this.saveAsSync(this.internal.location, secret);
-    }
-
-    private _savePrep(secret?: string): void {
-        if (!!secret) {
-            this.validateSecretKey(secret);
-        }
-
-        // make sure that all dispatch serviceIds still match services that are in the bot
-        for (const service of this.services) {
-            if (service.type === ServiceTypes.Dispatch) {
-                const dispatchService: IDispatchService = <IDispatchService>service;
-                const validServices: string[] = [];
-                for (const dispatchServiceId of dispatchService.serviceIds) {
-                    for (const this_service of this.services) {
-                        if (this_service.id === dispatchServiceId) {
-                            validServices.push(dispatchServiceId);
+            switch (service.type) {
+                case ServiceTypes.Dispatch:
+                    {
+                        let luisService = <ILuisService>service;
```

Comment:
const

Reply:
Note: TSLint is catching all of these and providing warnings. Please consider looking at those warnings and resolving them prior to commiting this code.